### PR TITLE
tools: kdumpid: include config.h before <dis-asm.h>

### PR DESCRIPTION
**Pull request by thesamesam**
Before including bfd.h (or any of the headers installed by binutils),
one has to define PACKAGE_NAME and PACKAGE_VERSION for kind of painful
reasons described in binutils bug PR14243.

This doesn't show up on some distros as they patch out the #ifdef check
in the headers, but it does on Gentoo where we don't, at least.

Moreover, we want to include config.h before *any* headers anyway, so
that e.g. AC_SYSTEM_EXTENSIONS and various other autoconf macros work
correctly if used.

Bug: https://sourceware.org/PR14243

### CODE REVIEW

**Comment by thesamesam** [comment]
The first commit is the truly essential one, the others are kind of future-proofing against someone getting a nasty surprise on trying to use an autoconf macro in future.

File: tests/testutil.h [hunk 1]
```diff
@@ -29,6 +29,8 @@
 #ifndef _TESTUTIL_H
 #define _TESTUTIL_H 1
 
+#include "config.h"
```
Comment (ptesarik) [inline tests/testutil.h:32]: In commit f2de47962860027889b773da05e876ea0e3d24f9 you keep `#include "config.h"` even in files that also include `testutil.h` and you provide a good explanation why that's a good idea. So is it really necessary to add this line here?

**Comment by ptesarik** [comment]
This PR looks good to me. Would you mind adding your `Signed-off-by:` to each of the three commits?

File: tests/testutil.h (hunk 1 above)
Comment (thesamesam) [reply tests/testutil.h:32]: I suspect I was in two minds about what you might prefer and ended up muddled :)

That said, `testutil.h` itself includes headers that could be affected by e.g. `AC_SYS_LARGEFILE` (`stdio.h`), so it makes sense from a perspective of ensuring all includes of standard headers are prefixed. But I can change it if you want as well.

**Comment by ptesarik** [comment]
Thank you for the fix, Sam!

**Pull request merged**